roles/nova-common/templates/etc/nova/nova.conf

```diff
@@ -124,8 +124,12 @@ rabbit_password={{ secrets.rabbit_password }}
 # Reserved Resources
 {% if ironic.enabled -%}
 reserved_host_memory_mb=0
-{% elif inventory_hostname in groups['controller'] -%}
-reserved_host_memory_mb=4096
+{% elif nova.reserved_host_memory_mb is defined %}
```

Comment:
That is just a blanket for something other than compute or controller.  Not sure what you mean it would conflict with Ironic as these are elif's.